### PR TITLE
Fix #1152: add deferred bedtime arrival scenario

### LINKED ISSUE #1152: test: add cross-plugin scenario test for go_to_bed deferral → arrival → bedroom lit
## Background

PR #1151 / PR #1149 added a presence gate to `go_to_bed`: if no one is home, the trigger defers until someone arrives (within the 1-hour window), avoiding `isSleepPrepActive=true` with an empty house.

The unit test in `manager_test.go` covers the deferral in isolation. However, the production-incident invariant — deferred `go_to_bed` → arrival → bedroom lit normally — has no cross-plugin integration coverage.

## What to add

A scenario test in `test/integration/` validating:

- **GIVEN:** No one home at `go_to_bed` time → trigger deferred, `isSleepPrepActive=false`
- **WHEN:** Someone arrives home (within the 1-hour window) → `go_to_bed` fires, `isSleepPrepActive=true` briefly
- **THEN:** Bedroom welcome scene is NOT suppressed on arrival (lighting plugin ran the scene before `isSleepPrepActive` was set)

Adjacent coverage exists in `scenario_sleephygiene_test.go` for the `isSleepPrepActive=true` suppression path.

Raised from Test Review on PR #1151.

### PR DESCRIPTION
**Pull request by NickBorgers**
Resolves #1152

## Summary
Add a cross-plugin integration scenario for the deferred go_to_bed path: empty house at bedtime defers sleep prep, arrival runs the bedroom scene while sleep prep is still false, and the next scheduled sleep hygiene tick starts sleep prep.

Add a narrow sleep hygiene test hook to drive the scheduled check without waiting a real minute.

## Test Plan
- make unit-tests
- make integration-tests
- make pre-commit

🤖 Generated by the AI assistant

### CODE REVIEW

**Comment by NickBorgers** [comment]
## Documentation Review

### Worth Considering

`docs/flows/SLEEP_HYGIENE.md` "Daily Schedule Triggers" diagram shows `go_to_bed` firing directly to `flashLights2` and `startSleep` with no condition. The fix adds an `isAnyoneHome` gate that defers the trigger rather than skipping it — the diagram now misrepresents the behavior. The Inputs table already lists `isAnyoneHome` and the code comment + tests explain the invariant well, so this is non-blocking. If you update it, replace the direct edges from `goToBed` with a decision node:

```mermaid
goToBed --> checkHome{"isAnyoneHome?"}
checkHome -->|Yes| flashLights2["Flash common area lights"]
checkHome -->|Yes| startSleep["Set musicPlaybackType = sleep"]
checkHome -->|No| defer["Defer: check next minute tick"]
```

Also non-blocking: `VISUAL_ARCHITECTURE.md` State Variable Dependency Graph is missing `AnyoneHome --> SleepHygiene` (the plugin was already polling `isAnyoneHome` before this PR; pre-existing gap made more notable by this change).

### Conclusion
✅ Approved — test-only PR with a one-line bug fix; no blocking doc gaps.

**Comment by NickBorgers** [comment]
## Code Review

### Blocking Issues

**`scenario_sleepprep_lighting_test.go:207-211` — GIVEN phase never exercises the defer code path**

The test is named `GoToBedDeferredUntilArrival` and the docstring says "GIVEN no one is home at go_to_bed time, sleep prep must defer," but `TriggerScheduledCheckForTest()` is never called while `isAnyoneHome=false`. That means the `if !isAnyoneHome { defer }` branch in `checkTimeTriggers` is never hit. The `waitForBoolState(..., "isSleepPrepActive", false, ...)` passes trivially because it is just asserting the initial value you set three lines above — no sleep hygiene logic has run at all.

The regression this test is meant to catch (removing the `isAnyoneHome` gate from `checkTimeTriggers`) would still pass as written, because `triggeredToday["go_to_bed"]` is empty when the single manual tick fires in the WHEN phase regardless of whether the defer branch exists.

**Fix:** Add a `sleepMgr.TriggerScheduledCheckForTest()` call in the GIVEN section (while `isAnyoneHome=false`), then assert `isSleepPrepActive` is still false using `require.Never` or a snapshot assert after `waitForProcessing`. That proves the defer branch ran and did not advance the trigger. The existing WHEN/THEN calls then validate that a second tick, now with someone home, fires correctly.

### Worth Considering

- `waitForBoolState` uses `require.Eventually` (passes as soon as the value matches), so using it for a "must remain false" invariant is weaker than `require.Never` — it cannot catch a brief true→false flip.

### Conclusion

⚠️ Needs changes — the test hook is fine, but the scenario test does not cover the deferred path it is named after.

File: homeautomation-go/test/integration/scenario_sleepprep_lighting_test.go [hunk 1]
```diff
@@ -169,3 +171,88 @@ func TestScenario_SleepPrepNotActive_LightingControlsBedroom(t *testing.T) {
...
+// THEN the next scheduled sleep hygiene tick may fire go_to_bed and set
+// isSleepPrepActive=true without having suppressed the arrival lighting.
+func TestScenario_GoToBedDeferredUntilArrival_BedroomWelcomeNotSuppressed(t *testing.T) {
+	t.Parallel()
+
+	server, client, stateManager, baseCleanup := setupTest(t)
+	defer baseCleanup()
+
+	logger := testlogger.New()
+	goToBedWindow := time.Date(2024, 1, 15, 23, 0, 0, 0, time.UTC)
```
Comment (NickBorgers) [inline homeautomation-go/test/integration/scenario_sleepprep_lighting_test.go:191]: The GIVEN assertion here confirms the value you just set three lines above — no sleep hygiene tick has run yet, so this does not prove the system deferred. Adding `sleepMgr.TriggerScheduledCheckForTest()` here (while `isAnyoneHome=false`) and then asserting `isSleepPrepActive` is still false would actually exercise the `if !isAnyoneHome` gate in `checkTimeTriggers` and make this test falsifiable.

**Review by NickBorgers — COMMENTED** [review]
Inline: the GIVEN waitForBoolState never exercises the defer path.

**Comment by NickBorgers** [comment]
## Design Review

**Alignment check:** PASS
**Scope check:** PASS

### Worth Considering

- This PR is stacked on `fix/sleep-prep-empty-house` (#1149), which is still open. It cannot land to `main` until #1149 merges; a reviewer scanning only this PR will miss that dependency.
- The `assert.False(t, isSleepPrep)` check (immediately after `waitForServiceCallWithEntitySince`) races theoretically with the 1-minute ticker — if the ticker fired between arrival and the assertion the test would false-positive. In practice this is safe at test speeds, but worth documenting with a comment that explains *why* the race window is acceptable.
- `TriggerScheduledCheckForTest` is consistent with `TriggerWakeForTest` and `TriggerBeginWakeForTest`, but none of those synchronize with the background timer goroutine. Running the test with `-race` could surface a data race on `m.triggeredToday` if the ticker fires at exactly the wrong moment. The test timeout makes this practically inert, but it is structurally the same pattern already present — not introduced by this PR.

### Conclusion
✅ Approved

File: homeautomation-go/internal/plugins/sleephygiene/manager.go [hunk 2]
```diff
@@ -1277,6 +1277,13 @@ func (m *Manager) TriggerWakeForTest() {
...
 }
 
+// TriggerScheduledCheckForTest runs the same scheduled trigger check that the
+// background timer runs once per minute.
+func (m *Manager) TriggerScheduledCheckForTest() {
+	m.logger.Info("Test: Triggering scheduled check directly")
+	m.checkTimeTriggers()
+}
+
 // TriggerBeginWakeForTest is a test helper that directly triggers the begin_wake sequence.
```
Comment (NickBorgers) [inline homeautomation-go/internal/plugins/sleephygiene/manager.go:1300]: `TriggerScheduledCheckForTest()` calls `checkTimeTriggers()` directly from the test goroutine while `Start()` has already launched `go m.runTimerLoop()` which calls the same function via a 1-minute ticker. Both paths read, write, and delete `m.triggeredToday` (a `map[string]time.Time`) with no mutex. This is a real data race.

Fix options:
1. Add a `sync.Mutex` to `Manager` protecting `triggeredToday`.
2. Send a signal to the timer goroutine (same pattern as `TriggerWakeForTest`).

**Review by NickBorgers — COMMENTED** [review]
TriggerScheduledCheckForTest() calls checkTimeTriggers() directly from the test goroutine while Start() has already launched go m.runTimerLoop() which calls the same function via a 1-minute ticker. Both paths read, write, and delete m.triggeredToday (a map[string]time.Time) with no mutex.

**Comment by NickBorgers** [comment]
## Concurrency Review

### Blocking Issues

**`triggeredToday` map race — `manager.go:1287`**

`TriggerScheduledCheckForTest()` calls `checkTimeTriggers()` directly from the test goroutine concurrently with the background `runTimerLoop()` goroutine (started by `Start()`), which calls the same function every 60s. `m.triggeredToday` is a `map[string]time.Time` with zero synchronization — no mutex anywhere on `Manager`. Both goroutines read, write, range, and delete from this map.

The practical hit rate is low (~one 1-second window per 60-second tick), but `-race` is enabled in `make unit-tests` and will flag it eventually, failing CI non-deterministically.

This pattern is distinct from `TriggerWakeForTest()`, which calls `handleWake()` — a function that does not touch `triggeredToday`. `TriggerScheduledCheckForTest` is the first test helper to call the full `checkTimeTriggers()` path, making this a new race introduced by this PR.

**Fix (prefer option 2 — minimal change):**
- Option 1: Add `sync.Mutex` to `Manager` and lock it inside `checkTimeTriggers()`, `resetTriggeredToday()`, and any other site that touches `triggeredToday`.
- Option 2: Have `TriggerScheduledCheckForTest()` send a signal to the timer goroutine (e.g. an unbuffered channel) and block until it processes — same pattern as other test helpers. This avoids introducing a mutex and keeps `checkTimeTriggers()` single-threaded.

### Worth Considering

The new unit test `TestCheckTimeTriggers_GoToBed_DeferredWhenEmpty` silently skips when `schedule_config.yaml` is not resolved at the expected relative path. Not a blocker since the integration test covers the behavior, but worth noting that the unit test provides no safety net in environments where the config path differs.

### Conclusion
⚠️ Needs changes — one real data race introduced by `TriggerScheduledCheckForTest`, low hit rate but `-race` will catch it eventually.

**Comment by NickBorgers** [comment]
## Test Review

✅ Approved — small, correct fix for sleep prep deferral when house is empty, with full unit + integration coverage and all 11 integration tests passing.

### Worth Considering
- The `../../../configs` path in the new integration test differs from `../../configs` used elsewhere in the same file; benign but a one-line comment would help future readers understand why this test requires schedule config pre-loaded.
- Edge case where someone leaves and never returns during the 1-hour window (go_to_bed simply skips that day) is intentional but unstated — a brief comment in the defer log path could help.

### Conclusion
✅ Approved

**Comment by NickBorgers** [comment]
## 🚦 Merge Decision

**Decision: 🟢 GO**

**Findings addressed:**

- **Blocking (Code Review) — GIVEN phase never exercises defer path:** Fixed. Added `sleepMgr.TriggerScheduledCheckForTest()` in the GIVEN section while `isAnyoneHome=false`, then used `require.False` directly after the synchronous call. This proves the defer branch ran and left `isSleepPrepActive=false`. The previous `waitForBoolState` only confirmed the initial setup value.

- **Blocking (Concurrency Review) — `triggeredToday` map race on `TriggerScheduledCheckForTest`:** Fixed with Option 2 (channel-based, minimal change). Added `triggerCheckCh chan chan struct{}` to Manager. `TriggerScheduledCheckForTest` now sends a done-channel to the timer goroutine and blocks until it receives; the timer goroutine calls `checkTimeTriggers()` and closes the done-channel. This keeps `checkTimeTriggers()` single-threaded and eliminates the race.

- **Worth Considering (Docs Review) — `SLEEP_HYGIENE.md` diagram missing `isAnyoneHome` gate:** Fixed. Replaced the direct `goToBed → flashLights2/startSleep` edges with a `checkHome{"isAnyoneHome?"}` decision node showing the defer path. Updated the Schedule Time Sources table description for `go_to_bed` to mention the deferral.

- **Worth Considering (Docs Review) — `VISUAL_ARCHITECTURE.md` missing `AnyoneHome → SleepHygiene` edge:** Accepted as pre-existing gap; out of scope for this PR.

- **Worth Considering (Code Review) — `waitForBoolState` weak for "must remain false":** Fixed as part of the GIVEN-phase fix — now using direct `require.False` after the synchronous channel call instead of `require.Eventually`.

- **Worth Considering (Design Review) — `assert.False` snapshot races theoretically with 1-minute ticker:** Added inline comment explaining why the race window is acceptable at test speed.

- **Worth Considering (Design Review) — PR stacked on #1149:** Noted; #1149 must merge before this lands on main.

- **Worth Considering (Test Review) — `../../../configs` path differs from `../../configs`:** Added one-line comment explaining the extra parent hop.

- **Worth Considering (Test Review) — leave-and-never-returns edge case unstated:** Added one-line comment in the defer log path in `manager.go`.

- **Worth Considering (Concurrency Review) — unit test silently skips when schedule config unavailable:** Accepted. The integration test covers the behavior; the unit test is a belt-and-suspenders check. Pre-existing pattern across the test suite.

**Applied fixes:** Channel-based `TriggerScheduledCheckForTest` (eliminates data race), GIVEN-phase explicit defer-branch exercise, SLEEP_HYGIENE.md diagram update, minor comments. All unit tests (74.5% coverage) and 12 integration tests pass with `-race`.

Ready for merge to main and production deployment (after #1149 lands).
